### ISSUE TITLE
a problem met in reimplmentation

### ISSUE THREAD

**Issue by russellllaputa** [ZhichengHuang/CMAE#8]
Hi, Thank you for your excellent work. 
We are working on reimplementing CMAE, but there is a very tricky problem we meet during pretraining. The infoNCE loss increases gradually while the pixel reconstruction loss goes down. The accuracy in Imagenet-1k is 83.4% after fine-tuning which is even worse than MAE.  We have carefully followed the configurations written in your paper.  Are there any details missing in the paper?  We will appreciate that if you can release the code or provide pretrained weights. Looking forward to your reply.

**Comment by ZhichengHuang** [comment]
Thank you for your message and for your interest in our work. We have consulted with our co-authors and have decided to open-source the code for our paper, following feedback from the peer review process. However, in the meantime, we can provide you with the pretraining weights for CMAE, which can be found at   https://cmae.s3.us-west-1.amazonaws.com/weight/cmae_vit_b16.pth . This checkpoint was pretraining for 1600 epochs, and we hope that it will be helpful to you.

**Comment by guoxih** [comment]
I appreciate the pretrained weight you provided, which does help a lot.
I am wondering why would the infoNCE loss go down first and increase gradually, is that how you designed it or does just my implementation have some unknown bugs? 
many thanks

**Comment by tungdq212** [comment]
Hello @russellllaputa, i try to reimplement this paper too, but not sure about my code. Can you help me? I will post my  question and code below.

1. Infer InfoNCE loss

```
def _constrastive_loss(student_output, teacher_output, temperature=0.07):
    # InfoNCE loss
    preds = student_output
    targets = teacher_output

    logits = preds @ targets.T / temperature
    labels = torch.arange(logits.shape[0]).long().to(logits.device)
    loss = F.cross_entropy(logits, labels, reduction="mean")
    return loss
```
2. 2 decoders
```
      # CMAE pixel decoder specifics
      self.pixel_decoder_blocks = nn.ModuleList([
          Block(decoder_embed_dim, decoder_num_heads, mlp_ratio, qkv_bias=True, norm_layer=norm_layer)
          for i in range(decoder_depth)])

      self.pixel_decoder_norm = norm_layer(decoder_embed_dim)
      self.pixel_decoder_pred = nn.Linear(decoder_embed_dim, patch_size**2 * in_chans, bias=True) # decoder to patch

      # CMAE feature decoder specifics
      self.feature_decoder_blocks = nn.ModuleList([
          Block(decoder_embed_dim, decoder_num_heads, mlp_ratio, qkv_bias=True, norm_layer=norm_layer)
          for i in range(decoder_depth)])

      self.feature_decoder_norm = norm_layer(decoder_embed_dim)
      self.feature_decoder_pred = nn.Linear(decoder_embed_dim, embed_dim, bias=True)
      self.prediction_head = Mlp(embed_dim, hidden_dim, embed_dim, norm_layer=nn.BatchNorm1d if use_bn else None)
```
3. Follow paper, I set decoder's depth=2 and num_heads=2 for Vit-B. Is it true?

**Comment by tungdq212** [comment]
@ZhichengHuang @russellllaputa
